src/client/proxy/http.cc

```diff
@@ -32,178 +32,232 @@
 
 #include "client/proxy/http.h"
 
+#include <atomic>
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/regex.hpp>
 #include <boost/lexical_cast.hpp>
 #include <boost/regex.hpp>
-
-#include <atomic>
+#include <boost/tokenizer.hpp>
 #include <cassert>
 #include <cstdint>
 #include <cstring>
+#include <map>
 #include <memory>
 #include <string>
+#include <utility>
 
+#include <boost/foreach.hpp>
 #include "client/api/streaming.h"
 #include "client/context.h"
 #include "client/destination.h"
 #include "client/tunnel.h"
 #include "client/util/http.h"
-
 #include "core/router/identity.h"
 
 #include "core/util/i2p_endian.h"
-
 namespace kovri {
 namespace client {
 
-//
-// Server
-//
-
-HTTPProxyServer::HTTPProxyServer(
+HTTPProxyServerService::HTTPProxyServerService(
     const std::string& name,
     const std::string& address,
     std::uint16_t port,
     std::shared_ptr<kovri::client::ClientDestination> local_destination)
     : TCPIPAcceptor(
           address,
           port,
-          local_destination ?
-            local_destination :
-            kovri::client::context.GetSharedLocalDestination()),
-      m_Name(name) {}
+          local_destination
+              ? local_destination
+              : kovri::client::context.GetSharedLocalDestination()),
+      m_Name(name) {
+}
 
 std::shared_ptr<kovri::client::I2PServiceHandler>
-  HTTPProxyServer::CreateHandler(
+HTTPProxyServerService::CreateHandler(
     std::shared_ptr<boost::asio::ip::tcp::socket> socket) {
   return std::make_shared<HTTPProxyHandler>(this, socket);
 }
 
-//
-// Handler
-//
-
-void HTTPProxyHandler::AsyncSockRead() {
+void HTTPProxyHandler::Handle() {
```

Comment:
I had no where else to comment, this comment was for your handling implementation as it relates to I2PService (and your handling implementation).